### DISCUSSION TITLE
Current API functions

### DISCUSSION THREAD

**Discussion by useronkel** [nilsteampassnet/TeamPass#3523] [Q&A] [answered]
Hi.
I hope the api is a 'work-in-progress'?
I'm missing a call like getItemDataForLabel() :-)
I've currently written a small extension on my own to get this feature.

In my opinion it doesn't make any sense to access the api to get an id from any kind of entry.
It is not possible to get an id for an entry, when they will be newly created.

I suggest, no one cares about an id, but a label is the way to go.
Currently I'm evaluating teampass and the database already has thousands of entries (~ 2500 per staging environment). 
You guess it....impossible to get a specific entry with an id as criteria.

**Comment by nilsteampassnet** [comment]
Hi @useronkel 
Thank you for the question.
Yes still in a WIP status. I will implement a search by label.
THanks

**Reply by nilsteampassnet** [reply]
Hi @useronkel 
I added 2 new features in latest commit.

1. Get items by `label` : https://documentation.teampass.net/#/api-basic?id=get-item-data-by-label
2. Get items by `description` : https://documentation.teampass.net/#/api-basic?id=get-item-data-by-description

I have added the possibility to use % symbol to open/refine the search.

Both return an array of items fetching the search

**Comment by useronkel** [comment, marked as answer]
Great job!
Thx a lot again for your enthusiasm :-)
